### DISCUSSION TITLE
What support does Skywalking have for MongoDB?

### DISCUSSION THREAD

**Discussion by andrewgkew** [apache/skywalking#7545] [Q&A] [answered]
I found an issue and merge where someone in the community kindly added support for MongoDB

https://github.com/apache/skywalking/pull/148

My application is using MongoDB and when I add it to Skywalking I see the MongoDB icon in the Topology and I get very simple stats on the DB tab in the UI what exactly does this new feature allow for as I expected either to need to add a sidecar proxy on my mongo pod but it seems to do it out of the box?

Thanks

**Comment by wu-sheng** [comment]
The PR you referred was 4 years ago, it added a tracing plugin for MongoDB Java client.
What do you expect? I am not sure the question.

**Comment by wu-sheng** [comment]
What do you mean about sidecar?

**Reply by andrewgkew** [reply]
thanks @wu-sheng  - @kezhenxu94  answered my question below, I was looking to see if there was some sort of agent for MongoDB I could run as a sidecar to the pod, but I now understand that the feature is client side and part of the agent (Java) and there is no server side instrumentation for mongodb. Thanks

**Comment by kezhenxu94** [comment, marked as answer]
The feature is to intercept client side MongoDB driver API calls and generate the topology and metrics, so there is no (MongoDB) server side instrumentation and you don't need sidecar there, it works out of box as long as you add the agent to your services which accessing the db with supported driver version.

**Reply by andrewgkew** [reply]
Thanks @kezhenxu94 - thats exactly what I was after but didnt explain myself properly, the fact that the feature is a client side agent at intercepts the drive API calls. It does work out of the box just wanted to check if there was anything more to it and if I got any more stats, but think my answer is no. Thank you

**Reply by wu-sheng** [reply]
All metrics are generated from trace, and powered by OAL script.
If you want to add new, it is totally customizable.

**Reply by andrewgkew** [reply]
Thanks @wu-sheng - any docs on how I can customize this please?

**Reply by wu-sheng** [reply]
You need to understand trace first, then use this.

https://skywalking.apache.org/docs/main/v8.7.0/en/guides/backend-oal-scripts/

https://skywalking.apache.org/docs/main/v8.7.0/en/concepts-and-designs/oal/
